packages/Angular/explorer-core/src/lib/list-component/list-component.ts

```diff
@@ -0,0 +1,200 @@
+  public title: string = ''
+
+  constructor(private notificationService: NotificationService, private router: Router) {
+    this._contextUser = null
+   }
+
+  get getEntities() {
+    return this.entities
+  }
+
+  get getList() {
+    return this.list
+  }
+
+  private appendObj(listObj: List) {
+    const transformed = {
+      ID: listObj.ID,
+      Name: listObj.Name,
+      Description: listObj.Description,
+      Entity: listObj.Entity,
+      EntityID: listObj.EntityID,
+      User: listObj.User
+    }
+    if(this.selectedRow) {
+      const index: number = this.list.findIndex((item) => item.ID === listObj.ID)
+      Object.assign(this.list[index], transformed)
+    }
+    else this.list = [ ...this.list, transformed]
+       
+  }
+
+  private updateList(listID: number) {
+    this.list = this.list.filter((item) => item.ID !== listID)
+  }
+
+  private resetForm() {
+    this.listName = ''
+    this.listDescription = ''
+    this.selectedEntities = []
+    this.selectedRow = null
+  }
+
+  async ngOnInit(): Promise<void> {
+    await this.loadLists()
+    await this.loadEntities()
+  }
+
+  public async loadLists(): Promise<void> {
+    this.loader = true
+    const rv = new RunView();
+    const models = await rv.RunView({EntityName: 'Lists', ResultType: 'entity_object'}, this.md.CurrentUser);
+    this.list = models?.Results.map((entity) => {
+    return {
+      ID: entity.ID,
+      Name: entity.Name,
+      Description: entity.Description,
+      Entity: entity.Entity,
+      EntityID: entity.EntityID,
+      User: entity.User,
+      EntityInfo: entity.EntityInfo
+    }
+    });
+    this.loader = false;
+  }
+
+  public async loadEntities(): Promise<void> {
+    const rv = new RunView();
+    const models = await rv.RunView({EntityName: 'Entities', ResultType: 'entity_object'}, this.md.CurrentUser);
+    this.entities = models?.Results.map((entity) => {
+    return {
+      id: entity.ID,
+      name: entity.Name
+    }
+  });
+  }
+
+  public async saveList(): Promise<void> {
+    this.btnLoader = true
+    const md = new Metadata();
+    const listPromise = md.GetEntityObject<ListEntity>('Lists');
+    const list = await listPromise;
+    if(this.selectedRow?.ID) 
+      await list.Load(this.selectedRow.ID)
+    else 
+      list.NewRecord();
+    list.Name = this.listName;
+    list.Description = this.listDescription;
+    list.UserID = md.CurrentUser.ID;
+    list.EntityID = this.selectedEntities[0].id;
+
+    try {
+        await list.Save();
+        this.appendObj(list)
+        this.btnLoader = false;
+        this.opened = false
+        this.resetForm()
+    } catch (error) {
+      console.log('Error:', error)
+    }
+  }
+
+  public async deleteEntity() {
+      this.btnLoader = true;
+      const md = new Metadata();
+      const listObj = <ListEntity>await md.GetEntityObject('Lists');
+      await listObj.Load(this.selectedEntity?.ID); // load the view to be deleted
+      const response = await listObj.Delete()
```

Comment:
Before you delete the list, you need to check if there record has any dependencies first. otherwise the delete will fail. To check if a record has any dependencies, you do 
`const md: Metadata = new MetaData()`
`let dep: RecordDependency[] = await md.GetRecordDependencies(this.selectedEntity.EntityInfo.Name, this.selectedEntity.CompositeKey); `
`if(dep.length > 0) return;`